src/rest.js

```diff
@@ -30,6 +40,28 @@ export default class RestClient {
         ...config.headers,
       },
     }));
+
+    if (notifications) {
+      this.notifications = new NotificationBus(this, notificationBusOpts);
```

Comment:
I never really liked the bidirectional dependency here.  I wonder if it would be better to have `login`/`logout` events on this class to decouple the two would be better, e.g.
```javascript
class NotificationBus {
  constructor($rest, ...) {
    ...
    $rest.$on('login', this.connect);
    $rest.$on('logout', this.disconnect);
  }
}
```
These events could be used as an alternative to watching the `user` attribute directly.  Not that we would want to encourage doing so, but there are cases where it is hard to avoid.

It would make this class directly depend on Vue (or some alternate pubsub), but I'm not convinced that's necessarily a big problem.

Reply:
I had also thought of that architecture, and I think it's better than the currently written one. For this PR, I may just back out cbd9ce8 and we can decide on this in a future PR scoped to just that topic.